hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/sigs/Rationalization.java

```diff
@@ -113,6 +114,7 @@ void resetFor(final SwirldsTxnAccessor txnAccessor) {
 
         pkToSigFn.resetAllSigsToUnused();
         bodySigningFactory.resetFor(txnAccessor);
+        txnAccessor.setPendingCompletions(Collections.emptyList());
```

Comment:
Wow, ok, that's a great catch

Let me try to summarize what the issue was and see if I got it right:
* We receive a couple of hollow finalization transactions for the same account at the same time
* During pre-handle, they all (or at least 2 of them) set a `PendingCompletion` in the txn accessor for the hollow account
* The first transaction that gets handled finalizes the account
* The subsequent transaction(s) that have a `PendingCompletion` see that the hollow account has changed since pre-handle via the `linkedRefs`, so they perform a `Rationalization`. In there, we re-do all the signature checks, the `HollowScreening` does nothing, since the hollow account will have a valid `JECDSA...Key `now. **However, we do not clear the pending completions set during pre-handle. Therefore, even if they are no longer valid, `HollowAccountFinalziationLogic` will still execute them.**

And the `hollowAccountOccursFinalizationOnlyOnceWhenMultipleFinalizationTensComeInAtTheSameTime` E2E (whose name I really managed to butcher, btw :smile: ) wasn't running into this scenario all the time, since the subsequent transactions may have reached preHandle only after the first transaction was handled, thus not setting any pending completions at all in the first place

Does that track?

Reply:
Seems exactly right @dimitar-dinev! This was an excellent EET, I guess it just needed a few more opportunities to run. 😁

Reply:
So maybe not that excellent after all 😄 Will think about more corner test scenarios we may be missing in the finalization suite